src/AspNet.Security.OAuth.Bitrix24/Bitrix24AuthenticationOptions.cs

```diff
@@ -0,0 +1,85 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using System.Globalization;
+using System.Security.Claims;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.OAuth;
+using static AspNet.Security.OAuth.Bitrix24.Bitrix24AuthenticationConstants;
+
+namespace AspNet.Security.OAuth.Bitrix24
+{
+    /// <summary>
+    /// Defines a set of options used by <see cref="Bitrix24AuthenticationHandler"/>.
+    /// </summary>
+    public class Bitrix24AuthenticationOptions : OAuthOptions
+    {
+        private string? tenantHostName;
+
+        public string? TenantHostName
+        {
+            get
+            {
+                return tenantHostName;
+            }
+
+            set
+            {
+                tenantHostName = value;
+                AuthorizationEndpoint = string.Format(CultureInfo.InvariantCulture, Bitrix24AuthenticationDefaults.AuthorizationEndpoint, value);
+                TokenEndpoint = string.Format(CultureInfo.InvariantCulture, Bitrix24AuthenticationDefaults.TokenEndpoint, value);
+                UserInformationEndpoint = string.Format(CultureInfo.InvariantCulture, Bitrix24AuthenticationDefaults.UserInformationEndpoint, value);
+            }
+        }
+
+        public Bitrix24AuthenticationOptions()
+        {
+            ClaimsIssuer = Bitrix24AuthenticationDefaults.Issuer;
+            CallbackPath = Bitrix24AuthenticationDefaults.CallbackPath;
+
+            AuthorizationEndpoint = Bitrix24AuthenticationDefaults.AuthorizationEndpoint;
+            TokenEndpoint = Bitrix24AuthenticationDefaults.TokenEndpoint;
+            UserInformationEndpoint = Bitrix24AuthenticationDefaults.UserInformationEndpoint;
+
+            ClaimActions.MapJsonKey(ClaimTypes.NameIdentifier, UserFields.Id);
```

Comment:
Mapping everything is going to create a very large authentication cookie storing all of these claims.

Are they really all needed by default?

The normal pattern is to only automatically configure the basic/required claims, and users can map additional ones themselves in their individual configuration.